CMakeLists.txt

```diff
@@ -380,241 +364,342 @@ if(${PREFIX}ENABLE_CODESIGN AND (EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/cmake/codes
+						set(OBS_DOWNLOAD_HASH "CA19E8260E1A556E6231D75064837C61C6D480BD90C97A0B930005AE527BF625")
+					endif()
 				endif()
-			else()
-				message(FATAL_ERROR "${LOGPREFIX} Platform '${D_PLATFORM_OS}' with architecture '${D_PLATFORM_ARCH}' and bitness '${D_PLATFORM_BITS}' is not supported.")
-				return()
-			endif()
-		elseif(D_PLATFORM_MAC)
-			if (D_PLATFORM_ARCH_X86)
-				set(_DOWNLOAD_OBS_URL "https://github.com/Xaymar/obs-studio/releases/download/${_DOWNLOAD_OBS_VERSION}/obs-studio-x64-0.0.0.0-macos-${D_PLATFORM_ARCH}-${D_PLATFORM_BITS}.7z")
-				if (D_PLATFORM_BITS EQUAL 64)
-					set(_DOWNLOAD_OBS_HASH "SHA256=F15BC4CA8EB3F581A94372759CFE554E30D202B604B541445A5756B878E4E799")
-				else()
-					message(FATAL_ERROR "${LOGPREFIX} Platform '${D_PLATFORM_OS}' with architecture '${D_PLATFORM_ARCH}' and bitness '${D_PLATFORM_BITS}' is not supported.")
-					return()
+			elseif(D_PLATFORM_MAC)
+				if (D_PLATFORM_ARCH_X86)
+					if (D_PLATFORM_BITS EQUAL 64)
+						set(OBS_DOWNLOAD_URL "https://github.com/Xaymar/obs-studio/releases/download/27.2.4-ci/macos-x86_64.tar.gz")
+						set(OBS_DOWNLOAD_HASH "0DD5A57DD537B97CAA8470212F8F6B637F47D4742A5D1F17787F4FE9DBC70B33")
+					endif()
 				endif()
-			else()
-				message(FATAL_ERROR "${LOGPREFIX} Platform '${D_PLATFORM_OS}' with architecture '${D_PLATFORM_ARCH}' and bitness '${D_PLATFORM_BITS}' is not supported.")
+			endif()
+
+			# Verify that the platform, architecture and bitness is supported.
+			if(OBS_DOWNLOAD_URL STREQUAL "")
+				message(FATAL_ERROR "${LOGPREFIX} Download for libOBS failed, as Platform '${D_PLATFORM_OS}' with architecture '${D_PLATFORM_ARCH}' and bitness '${D_PLATFORM_BITS}' is not supported.")
 				return()
 			endif()
 		else()
-			message(FATAL_ERROR "${LOGPREFIX} Platform '${D_PLATFORM_OS}' with architecture '${D_PLATFORM_ARCH}' and bitness '${D_PLATFORM_BITS}' is not supported.")
-			return()
+			set(OBS_DOWNLOAD_URL "${${PREFIX}OBS_DOWNLOAD_URL}")
+			set(OBS_DOWNLOAD_HASH "${${PREFIX}OBS_DOWNLOAD_HASH}")
 		endif()
-	endif()
 
-	# Download libOBS
-	download_project(
-		PROJ libobs
-		URL "${_DOWNLOAD_OBS_URL}"
-		URL_HASH "${_DOWNLOAD_OBS_HASH}"
-		DOWNLOAD_NO_PROGRESS OFF
-		UPDATE_DISCONNECTED OFF
-	)
+		if(OBS_DOWNLOAD_HASH STREQUAL "")
+			download_project(
+				PROJ libobs
+				PREFIX "autodeps"
+				URL "${OBS_DOWNLOAD_URL}"
+				DOWNLOAD_NO_PROGRESS OFF
+				UPDATE_DISCONNECTED OFF
+			)
+		else()
+			download_project(
+				PROJ libobs
+				PREFIX "autodeps"
+				URL "${OBS_DOWNLOAD_URL}"
+				URL_HASH "SHA256=${OBS_DOWNLOAD_HASH}"
+				DOWNLOAD_NO_PROGRESS OFF
+				UPDATE_DISCONNECTED OFF
+			)
+		endif()
+		CacheSet(${PREFIX}OBS_PATH "${libobs_SOURCE_DIR}")
+	endif()
+	
+	if(NOT EXISTS "${${PREFIX}OBS_PATH}/cmake/LibObs/LibObsConfig.cmake")
+		message(FATAL_ERROR "${LOGPREFIX} The provided path for libOBS is invalid as it did not contain '/cmake/LibObs/LibObsConfig.cmake'.")
+		return()
```

Comment:
Builds from the `master` branch of OBS don't ship this file anymore, is it possible to check if the headers are present instead?

Reply:
No, this file is required as it sets up everything. If your builds do not ship this anymore, then your builds are broken and should be fixed.

Reply:
I fixed the OBS build upstream, the CMake config files will be included in the OBS 28 Flatpak. Do you plan on releasing v0.12 before it is released?

Reply:
Also, OBS CMake (at least on the `master` branch) generates a `$PREFIX/cmake/libobs/libobsConfig.cmake`, not a `$PREFIX/cmake/LibObs/LibObsConfig.cmake` one (different capitalization).

Reply:
The planned release for 0.12 is somewhere around end of July/start of August. Exact dates are not set, since StreamFX is not a net-positive product and as such is more of a "when I have time" thing.

Reply:
> Also, OBS CMake (at least on the master branch) generates a $PREFIX/cmake/libobs/libobsConfig.cmake

StreamFX builds against tagged versions, and the change is for 27.2.4. 27.2.4 exports a "LibObsConfig.cmake", not a "libobsConfig.cmake". I do not know how you are building your CMake packages, but something is clearly wrong.

Reply:
> The planned release for 0.12 is somewhere around end of July/start of August. Exact dates are not set, since StreamFX is not a net-positive product and as such is more of a "when I have time" thing.

No worries, a time frame and knowing that you don't align to OBS releases is enough to give me an indication.

> StreamFX builds against tagged versions, and the change is for 27.2.4. 27.2.4 exports a "LibObsConfig.cmake", not a "libobsConfig.cmake". I do not know how you are building your CMake packages, but something is clearly wrong.

Yes, I was talking about the `master` branch, which will become OBS 28, so that you're aware of the change. Also targets will be prefixed with the `OBS::` namespace.

By the way, I'm not building OBS myself, I'm just grabbing the official builds from upstream CI.

Reply:
As far as I can tell, upstream OBS uses the same command I use: [OBS](https://github.com/obsproject/obs-studio/blob/master/CI/linux/03_package_obs.sh#L21) [Mine](https://github.com/Xaymar/obs-studio/blob/27.2.4-ci/.github/workflows/package.yml#L229). So whatever is adding namespaces and changing the name is a commit to master, and not present in 27.2.4.

Reply:
Yes, that's what I'm saying: the commit is https://github.com/obsproject/obs-studio/commit/df94b02075636dc0ace7b4d1e9706abeb64dfbbd, it is present in the `master` branch and will be present in OBS 28, and is not present in OBS 27.2.4. So when OBS 28 is released, you'll have to take the aforementioned changes into account to build against it.